src/core/mempool.rs

```diff
@@ -1129,13 +1167,110 @@ impl MemPoolDB {
         Ok(updated)
     }
 
+    /// Returns an iterator over the mempool entries that do have a fee rate, sorted by fee rate.
+    /// Page size is 10_000. TODO: Make this configurable.
+    fn sorted_fee_rate_transactions(connection: &Connection) -> TransactionPageCursor {
+        let sql = "
+        SELECT txid, origin_nonce, origin_address, sponsor_nonce, sponsor_address, fee_rate
+        FROM mempool
+        WHERE fee_rate IS NOT NULL
+        ORDER BY fee_rate DESC
+        LIMIT ?
+        OFFSET ?
+        ";
+        TransactionPageCursor {
+            connection,
+            base_query: sql.to_string(),
+            current_offset: 0,
+            page_size: 10_000,
+            current_page_remaining: vec![],
+        }
+    }
+
+    /// Take a batch of transactions *without* a fee rate estimate, in *arbitrary* order.
+    /// Page size is 10_000. TODO: Make this configurable.
+    /// Note: Nulls in the mempool are, up to a limit, over-written between mempool runs.
+    fn null_fee_rate_transactions(connection: &Connection) -> TransactionPageCursor {
```

Comment:
@jcnelson @obycode 

Important question: Should we introduce a sorting order for "null fee rate transactions"? We could sort by `accept_time` or `height`. These have indices already I think:

```
    "CREATE INDEX IF NOT EXISTS by_height ON mempool(height);",
    "CREATE INDEX IF NOT EXISTS by_timestamp ON mempool(accept_time);",
```

Reply:
Yeah, that's a good point. It seems like first come, first served makes sense here. What does `height` in this table actually mean?

Reply:
Yes -- if you're ordering transactions by these columns, then they need to be indexed.  I would check the query with `EXPLAIN QUERY PLAN` just to be sure.  Also, depending on the ultimate structure of the `WHERE` and `ORDER BY` clauses, you may need to create a compound index, such as:

```
CREATE INDEX IF NOT EXISTS by_height_and_ts ON mempool(height,accept_time);
```

Reply:
at `5dd7fe1140893a6e26bb28dd6c2fbac98f6d61e1` the query plans are

```
sqlite> EXPLAIN QUERY PLAN SELECT txid, origin_nonce, origin_address, sponsor_nonce, sponsor_address, fee_rate FROM mempool WHERE fee_rate IS NULL ORDER BY accept_time ASC;
QUERY PLAN
|--SEARCH TABLE mempool USING INDEX by_fee_rate (fee_rate=?)
`--USE TEMP B-TREE FOR ORDER BY
sqlite> EXPLAIN QUERY PLAN SELECT txid, origin_nonce, origin_address, sponsor_nonce, sponsor_address, fee_rate FROM mempool WHERE fee_rate IS NOT NULL ORDER BY fee_rate DESC;
QUERY PLAN
`--SCAN TABLE mempool USING INDEX by_fee_rate
```

Reply:
schema i have is as follows.. looks like there is an index on `accept_time`:

```
sqlite> .schema mempool
CREATE TABLE mempool(
        txid TEXT NOT NULL,
        origin_address TEXT NOT NULL,
        origin_nonce INTEGER NOT NULL,
        sponsor_address TEXT NOT NULL,
        sponsor_nonce INTEGER NOT NULL,
        tx_fee INTEGER NOT NULL,
        length INTEGER NOT NULL,
        consensus_hash TEXT NOT NULL,
        block_header_hash TEXT NOT NULL,
        height INTEGER NOT NULL,    -- stacks block height
        accept_time INTEGER NOT NULL,
        tx BLOB NOT NULL, last_known_origin_nonce INTEGER, last_known_sponsor_nonce INTEGER, fee_rate NUMBER,
        PRIMARY KEY (txid),
        UNIQUE (origin_address, origin_nonce),
        UNIQUE (sponsor_address,sponsor_nonce)
    );
CREATE INDEX by_txid ON mempool(txid);
CREATE INDEX by_height ON mempool(height);
CREATE INDEX by_txid_and_height ON mempool(txid,height);
CREATE INDEX by_sponsor ON mempool(sponsor_address, sponsor_nonce);
CREATE INDEX by_origin ON mempool(origin_address, origin_nonce);
CREATE INDEX by_timestamp ON mempool(accept_time);
CREATE INDEX by_chaintip ON mempool(consensus_hash,block_header_hash);
CREATE INDEX by_fee_rate ON mempool(fee_rate);
```

Reply:
obycode says he has this for the query plan... not sure why my db is different.. I got my db by 1) start with snapshot, 2) run the `master->develop` branch 

```
sqlite> explain query plan
   ...> SELECT txid, origin_nonce, origin_address, sponsor_nonce, sponsor_address, fee_rate
   ...>             FROM mempool
   ...>             WHERE fee_rate IS NOT NULL
   ...>             ORDER BY fee_rate DESC
   ...> ;
QUERY PLAN
`--SEARCH mempool USING INDEX by_fee_rate (fee_rate>?)
```